packages/manager/modules/web-universe-components/src/order/order.controller.js

```diff
@@ -0,0 +1,267 @@
+      serviceNameToAddProduct: this.serviceNameToAddProduct,
+      type: this.type,
+      user: this.user,
+    };
+
+    this.eventHandlers = {
+      onBeforePricingGetPlancode: this.onBeforePricingGetPlancode,
+      onCheckoutSuccess: this.onCheckoutSuccess,
+      onError: this.onError,
+      onGetConfiguration: this.onGetConfiguration,
+    };
+
+    this.loading = {
+      getCheckoutInformations: false,
+      validateCheckout: false,
+    };
+  }
+
+  getPricings() {
+    this.pricing = null;
+    this.planCode = this.eventHandlers.onBeforePricingGetPlancode();
+
+    if (!this.planCode) {
+      throw new Error('WucOrderComponent: Invalid plan code');
+    }
+
+    const catalogPricings = get(
+      this.bindings.catalog[
+        CATALOG_ITEM_TYPE_NAMES[this.bindings.catalogItemTypeName]
+      ].find(({ planCode }) => planCode === this.planCode),
+      'pricings',
+    );
+
+    if (!catalogPricings) {
+      throw new Error(
+        `WucOrderComponent: No pricing found for ${this.planCode}`,
+      );
+    }
+
+    this.pricings = this.computePricing(catalogPricings);
+
+    if (this.hasUniquePricing()) {
+      this.currentIndex += 1;
+      [this.pricing] = this.pricings;
+    }
+  }
+
+  createNewCart() {
+    return this.WucOrderCartService.createNewCart(this.user.ovhSubsidiary).then(
+      (cart) => {
+        this.cartId = cart.cartId;
+        return this.WucOrderCartService.assignCart(this.cartId);
+      },
+    );
+  }
+
+  computePricing(catalogPricings) {
+    const pricingsOfType = OrderController.filterPricingsByCapacities(
+      catalogPricings,
+      this.type,
+    );
+
+    let pricings = pricingsOfType;
+
+    if (this.type === ORDER_TYPES.RENEW) {
+      const pricingWithOnlyInstallationCapacity = OrderController.getUniqueInstallationPricing(
+        catalogPricings,
+      );
+      const pricingsWithOnlyRenewCapacity = pricingsOfType.filter(
+        ({ capacities }) => isEqual(capacities, [ORDER_TYPES.RENEW]),
+      );
+      const pricingsWitRenewAndInstallationCapacities = pricingsOfType.filter(
+        ({ capacities }) =>
+          isEqual(capacities.sort(), [
+            ORDER_TYPES.INSTALLATION,
+            ORDER_TYPES.RENEW,
+          ]),
+      );
+
+      if (
+        !pricingsWitRenewAndInstallationCapacities &&
+        (!pricingWithOnlyInstallationCapacity && pricingsWithOnlyRenewCapacity)
+      ) {
+        throw new Error('WucOrderComponent: No installation pricing found');
+      }
+
+      pricings = pricingsWithOnlyRenewCapacity
+        .map((pricing) => ({
+          ...pricing,
+          price:
+            pricing.price +
+            get(pricingWithOnlyInstallationCapacity, 'price', 0),
+          tax: pricing.tax + get(pricingWithOnlyInstallationCapacity, 'tax', 0),
+        }))
+        .concat(pricingsWitRenewAndInstallationCapacities);
+    }
+
+    return sortBy(pricings, 'price');
+  }
+
+  isFreePricing() {
+    return get(this.pricing, 'price') === 0;
```

Comment:
Maybe a class could be indicated to handle this kind of logic

Reply:
Yes we should export the logic to another `class` !